### ISSUE TITLE
Add parameter.exampleValues property

### ISSUE THREAD

**Issue by MikeRalphson** [Mermade/widdershins#10]
Convert parameter definition into a schema and use openapi-sampler again to derive example values.

**Comment by MikeRalphson** [comment]
Better might be a `parameter.exampleValues` *object* with an initial property `json`, in case we want to extend this further with language specific formats.

Getting Pythons's indenting rules correct may be complex.

Also to consider is whether `body` ought to be in the parameters array at this point - it's probably passed differently in most languages / frameworks.

**Comment by MikeRalphson** [comment]
@bc-AlyssNoland - do you have any input into this at the design stage?

**Comment by bc-AlyssNoland** [comment]
@MikeRalphson I agree with your point of using json to make it more language agnostic. 

I've been comparing the output of widdershins to the swagger editor as that is also built with a flavor of JS. They will indicate the body parameter and then the schema within the body. 

![image](https://cloud.githubusercontent.com/assets/9373485/23315323/9544936c-fa7a-11e6-9ab6-501c007698f7.png)

![image](https://cloud.githubusercontent.com/assets/9373485/23315334/a378d9e8-fa7a-11e6-8c42-d89bef155248.png)

The ruby template I was working on looks fairly similar to the python one fwiw
```ruby
require 'rest-client'
require 'json'


{{? data.method === 'put'}}
result = RestClient.{{=data.method}} '{{=data.url}}', {content_type: :json, accept: :json}, headers: {X-Auth-Client => 'ClientID', X-Auth-Token => 'AuthToken'}, params:
  {
		{{? data.parameters.length-1 != index}}
		{{~ data.parameters :p:index}}{{=p.name}}:'{{=p.type}}'{{? index == 0 }},{{?}}
		{{~}}{{?}}
  }
{{?? data.method === 'post'}}
result = RestClient.{{=data.method}} '{{=data.url}}', {content_type: :json, accept: :json}, headers: {X-Auth-Client => 'ClientID', X-Auth-Token => 'AuthToken'}, params:
  {
		{{? data.parameters.length-1 != index}}
		{{~ data.parameters :p:index}}{{=p.name}}:'{{=p.type}}'{{? index == 0 }},{{?}}
		{{~}}{{?}}
  }
{{?? data.method === 'get' || 'delete'}}
result = RestClient.{{=data.method}} '{{=data.url}}', {content_type: :json, accept: :json}, headers: {X-Auth-Client => 'ClientID', X-Auth-Token => 'AuthToken'}
{{?}}

p JSON.parse(result)
```

**Comment by MikeRalphson** [comment]
@bc-AlyssNoland yes, the collapsible schema in Swagger-UI is nice, it's a pity as we're going through a static format such as markdown, we don't have the option to inject scripts like that. Widdershins lists the Body parameter with it's type name (from the #/definitions/name if present) and then shows an **example** of that schema (using `openapi-sampler`) in the right hand pane.

I wonder whether it might be worth having a switch to control whether the schema itself or the example is output? I think showing both (though useful) might take up too much room on the right hand side.

FYI I've just pushed some updates, check out the README for new `data` properties, including `queryString`, `allHeaders`, `queryParameters`, `headerParameters`, and `exampleValues`.  They should make writing code templates easier (if not any more readable).

You can now also specify a callback function to get control before each template is output, in case you need to alter the `data` object in a any way.

**Comment by MikeRalphson** [comment]
> I wonder whether it might be worth having a switch to control whether the schema itself or the example is output?

`-- raw` option added for this.

**Comment by MikeRalphson** [comment]
Included in v1.1.0
